[v1.18] CI: Integration Tests: pkg/loadbalancer/tests - TestScript/hybrid-dsr (ubuntu 24.04 arm)

The integration test `TestScript/hybrid-dsr.txtar` is failing intermittently on `ubuntu-24.04-arm64` environments.

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/loadbalancer/tests   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestScript (0.00s)
--- FAIL: TestScript/hybrid-dsr.txtar (8.55s)

// Snippet of failure seen in the workflow

    scripttest.go:72: (command "* cmp maps.actual maps.expected" failed, retrying in 500ms...)
        diff maps.actual maps.expected
        --- maps.actual
        +++ maps.expected
        @@ -1,14 +1,14 @@
        -BE: ID=1 ADDR=10.244.1.2:53/UDP STATE=active
        -BE: ID=2 ADDR=10.244.1.1:80/TCP STATE=active
        -REV: ID=1 ADDR=0.0.0.0:30782
        -REV: ID=2 ADDR=10.96.50.105:53
        -REV: ID=3 ADDR=0.0.0.0:30781
        -REV: ID=4 ADDR=10.96.50.104:80
        -SVC: ID=1 ADDR=0.0.0.0:30782/UDP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=NodePort+non-routable
        -SVC: ID=1 ADDR=0.0.0.0:30782/UDP SLOT=1 BEID=1 COUNT=0 QCOUNT=0 FLAGS=NodePort+non-routable
        -SVC: ID=2 ADDR=10.96.50.105:53/UDP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=ClusterIP+non-routable
        -SVC: ID=2 ADDR=10.96.50.105:53/UDP SLOT=1 BEID=1 COUNT=0 QCOUNT=0 FLAGS=ClusterIP+non-routable
        -SVC: ID=3 ADDR=0.0.0.0:30781/TCP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=NodePort+non-routable+dsr
        -SVC: ID=3 ADDR=0.0.0.0:30781/TCP SLOT=1 BEID=2 COUNT=0 QCOUNT=0 FLAGS=NodePort+non-routable+dsr
        -SVC: ID=4 ADDR=10.96.50.104:80/TCP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=ClusterIP+non-routable+dsr
        -SVC: ID=4 ADDR=10.96.50.104:80/TCP SLOT=1 BEID=2 COUNT=0 QCOUNT=0 FLAGS=ClusterIP+non-routable+dsr
        +BE: ID=1 ADDR=10.244.1.1:80/TCP STATE=active
        +BE: ID=2 ADDR=10.244.1.2:53/UDP STATE=active
        +REV: ID=1 ADDR=0.0.0.0:30781
        +REV: ID=2 ADDR=10.96.50.104:80
        +REV: ID=3 ADDR=0.0.0.0:30782
        +REV: ID=4 ADDR=10.96.50.105:53
        +SVC: ID=1 ADDR=0.0.0.0:30781/TCP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=NodePort+non-routable+dsr
        +SVC: ID=1 ADDR=0.0.0.0:30781/TCP SLOT=1 BEID=1 COUNT=0 QCOUNT=0 FLAGS=NodePort+non-routable+dsr
        +SVC: ID=2 ADDR=10.96.50.104:80/TCP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=ClusterIP+non-routable+dsr
        +SVC: ID=2 ADDR=10.96.50.104:80/TCP SLOT=1 BEID=1 COUNT=0 QCOUNT=0 FLAGS=ClusterIP+non-routable+dsr
        +SVC: ID=3 ADDR=0.0.0.0:30782/UDP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=NodePort+non-routable
        +SVC: ID=3 ADDR=0.0.0.0:30782/UDP SLOT=1 BEID=2 COUNT=0 QCOUNT=0 FLAGS=NodePort+non-routable
        +SVC: ID=4 ADDR=10.96.50.105:53/UDP SLOT=0 LBALG=undef AFFTimeout=0 COUNT=1 QCOUNT=0 FLAGS=ClusterIP+non-routable
        +SVC: ID=4 ADDR=10.96.50.105:53/UDP SLOT=1 BEID=2 COUNT=0 QCOUNT=0 FLAGS=ClusterIP+non-routable   
```

A failure was seen in [this workflow](https://github.com/cilium/cilium/actions/runs/22700062884)

Branch: `v1.18`
Environment: `Ubuntu 24.04 (ARM)`

--- a/pkg/loadbalancer/tests/testdata/hybrid-dsr.txtar
+++ b/pkg/loadbalancer/tests/testdata/hybrid-dsr.txtar
@@ -3,8 +3,10 @@
 # Start the test application
 hive start
 
-# Add TCP and UDP services
+# Add TCP and UDP services. Add TCP first and wait for it to be
+# reconciled for consistent ID allocation order.
 k8s/add service-tcp.yaml endpointslice-tcp.yaml
+db/cmp frontends frontends-tcp.table
 k8s/add service-udp.yaml endpointslice-udp.yaml
 db/cmp frontends frontends.table
 db/cmp services services.table
@@ -19,6 +21,11 @@ lb/maps-dump maps.actual
 Name          Source   TrafficPolicy   Flags
 test/echo-tcp k8s      Cluster
 test/echo-udp k8s      Cluster
+
+-- frontends-tcp.table --
+Address               Type         ServiceName    PortName   Status  Backends
+0.0.0.0:30781/TCP     NodePort     test/echo-tcp  http       Done    10.244.1.1:80/TCP
+10.96.50.104:80/TCP   ClusterIP    test/echo-tcp  http       Done    10.244.1.1:80/TCP
 
 -- frontends.table --
 Address               Type         ServiceName    PortName   Status  Backends

